File: app/src/main/java.backup/com/lumiyaviewer/lumiya/slproto/messages/DetachAttachmentIntoInv.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;
import java.util.UUID;

public class DetachAttachmentIntoInv extends SLMessage {
    public ObjectData ObjectData_Field = new ObjectData();

    public static class ObjectData {
        public UUID AgentID;
        public UUID ItemID;
    }

    public DetachAttachmentIntoInv() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 36;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleDetachAttachmentIntoInv(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 1);
        byteBuffer.put((byte) -115);
        packUUID(byteBuffer, this.ObjectData_Field.AgentID);
        packUUID(byteBuffer, this.ObjectData_Field.ItemID);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.ObjectData_Field.AgentID = unpackUUID(byteBuffer);
        this.ObjectData_Field.ItemID = unpackUUID(byteBuffer);
    }
}

File: Linkpoint/src/main/app_backup/com/lumiyaviewer/lumiya/slproto/messages/DetachAttachmentIntoInv.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import com.lumiyaviewer.lumiya.slproto.handler.SLMessageHandler
import com.lumiyaviewer.lumiya.slproto.types.UUID
import com.lumiyaviewer.lumiya.slproto.types.UUIDPool
import java.nio.ByteBuffer

class DetachAttachmentIntoInv : SLMessage {
    var ObjectData_Field: ObjectData = ObjectData()

    class ObjectData {
        var AgentID: UUID = UUIDPool.ZeroUUID
        var ItemID: UUID = UUIDPool.ZeroUUID
    }

    init {
        this.zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 36
    }

    override fun Handle(sLMessageHandler: SLMessageHandler) {
        // sLMessageHandler.HandleDetachAttachmentIntoInv(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        byteBuffer.putShort(-1)
        byteBuffer.put(1.toByte())
        byteBuffer.put((-115).toByte())
        packUUID(byteBuffer, this.ObjectData_Field.AgentID)
        packUUID(byteBuffer, this.ObjectData_Field.ItemID)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        this.ObjectData_Field.AgentID = unpackUUID(byteBuffer)
        this.ObjectData_Field.ItemID = unpackUUID(byteBuffer)
    }
}
